File: CodeForces/1975C_Chamo_and_Mochas_Array.py
""" ******************************
Username: JahidHasan90
Problem: https://codeforces.com/problemset/problem/1975/C
Give Time to Time , It can hill everything.
*******************************"""


tc = int(input())


while tc:
    n = int(input())
    a = list(map(int,input().split()))
    ans=[]
    for i in range(n-1):
        ans.append(min(a[i],a[i+1]))
    
    for i in range(n-2):
        ans.append(min(a[i],a[i+2]))
    print(max(ans))

    tc-=1

File: CodeForces/1975C_Chamo_and_Mochas_Array.cpp
#include<bits/stdc++.h>
using namespace std;


/* ******************************
Username: JahidHasan90
Problem: https://codeforces.com/problemset/problem/1975/C
Give Time to Time , It can hill everything.
********************************/


int main(){
    int tc; cin>>tc;
    while (tc--)
    {
        int n;
        cin >> n;
        vector<int> a(n);
        for(int i=0;i<n;i++) cin >> a[i];
        vector<int> ans;
        for (int i = 0; i < n-1; i++)
        {
            ans.push_back(min(a[i],a[i+1]));
        }
        for (int i = 0; i < n-2; i++)
        {
            ans.push_back(min(a[i],a[i+2]));
        }
        cout << *max_element(ans.begin(), ans.end()) << endl;
    }
    return 0;
}


// g++ learn.cpp && ./a.exe